grav/withdraw/signaler.go

```diff
@@ -0,0 +1,63 @@
+package withdraw
+
+import (
+	"sync/atomic"
+)
+
+// Signaler allows a connection to be notified about a withdraw event and report
+// back to the hub that a withdraw has completed (using the WithdrawChan and DoneChan, respectively)
+type Signaler struct {
+	withdrawChan  chan bool
+	doneChan      chan bool
+	selfWithdrawn atomic.Value
+	peerWithdrawn atomic.Value
+}
+
+// NewSignaler creates a new Signaler based on the provided context
+func NewSignaler() *Signaler {
+	w := &Signaler{
+		withdrawChan:  make(chan bool, 1),
+		doneChan:      make(chan bool, 1),
```

Comment:
Will we ever signal a false value or is this just to block until the signal is sent. If it's the latter would it be more appropriate to use an empty sturct?

Reply:
agree with @rnpridgeon , if it's "wait until a signal, contents whatever" then let's use `chan struct{}`. 

See https://medium.com/@matryer/golang-advent-calendar-day-two-starting-and-stopping-things-with-a-signal-channel-f5048161018, https://dave.cheney.net/2014/03/25/the-empty-struct, and https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface